### PR TITLE
Raycast distance check

### PR DESCRIPTION
**Pull request by Bod9001**
there is some stuff that abuses the Raycast  causing mega lag, 
this puts a hard limit on it, so it can't be abused and logs in error,
found out that sounds also have a daggy range check, so optimiseed that as well

### CODE REVIEW

File: UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs [hunk 1]
```diff
@@ -340,6 +340,13 @@ private static bool IsInMatrix(Vector3Int worldPos, bool isServer, MatrixInfo ma
 			WorldTo = Worldorigin + (Vector3) (direction.normalized * distance);
 		}
 
+		if ((WorldTo - Worldorigin).Value.magnitude > 25)
+		{
+			Logger.LogError(" Limit exceeded on raycast, Look at stack trace for What caused it "); //Meant to catch up stuff that's been naughty and doing stuff like 900 tile Ray casts
+			return new CustomPhysicsHit();
+		}
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs]: Isnt the distance parameter also used to control the range? Also 25 seems maybe too small

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs]: 25 is about off screen and little bit more, on the edges up and below it plenty extra

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs]: What about the distance parameter?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs]: Distance parameter is just used to , automatically convert it into a Line  cast from a Raycast

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs]: Could probably allow it to go through but just log the magnitude?

**Review by NoooneyDude — DISMISSED** [review]

**Comment by Bod9001** [comment]
but lag tho?

File: UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs]: but lag tho?

**Comment by ThatDan123** [comment]
Might be good to stick the magnitude value into the log

**Comment by corp-0** [comment]
I also would let it pass and log the abuse, it will save someone in the future not knowing why their features that requires a longer ray doesn't work

**Comment by Aranclanos** [comment]
Dont let it pass, its lagging the server at times making it uplayable. In fact i'd suggest on merging this and make the log better in a different PR so we can get this fix in a build faster.

**Review by corp-0 — APPROVED** [review]

**Review by NoooneyDude — APPROVED** [review]

**Pull request merged**